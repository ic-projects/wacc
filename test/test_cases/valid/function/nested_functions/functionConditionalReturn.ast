-- Test: functionConditionalReturn.wacc

-- Uploaded file: 
---------------------------------------------------------------
# program has function which only contains an if statement and a return in each branch

# Output:
# true

# Program:

begin
  bool f() is 
    if true then
      return true
    else
      return false
    fi
  end
  bool x = call f();
  println x
end
---------------------------------------------------------------

-- Compiler Output:
-- Compiling...
-- Printing AST...
functionConditionalReturn.ast contents are:
===========================================================
0	Program
1	  - bool f()
2	    - IF
3	      - CONDITION
4	        - true
5	      - THEN
6	        - RETURN
7	          - true
8	      - ELSE
9	        - RETURN
10	          - false
11	  - int main()
12	    - DECLARE
13	      - TYPE
14	        - bool
15	      - LHS
16	        - x
17	      - RHS
18	        - f
19	    - PRINTLN
20	      - x
===========================================================
-- Finished

