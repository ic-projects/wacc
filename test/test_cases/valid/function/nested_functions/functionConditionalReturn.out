-- Compiling...
-- Assembling and Linking...
-- Executing...
===========================================================
true
===========================================================
The exit code is 0.
-- Finished
